source/gridfs/gridfs-spec.md

```diff
@@ -649,6 +649,32 @@ orphaned chunks with files_id equal to id before raising the error.
 
 If a networking or server error occurs, drivers MUST raise an error.
 
+### File deletion by filename
+
+To rename all revisions of a stored file with the specified filename, drivers SHOULD provide the method
+`delete_by_name`:
+
+```javascript
+class GridFSBucket {
+
+  /**
+   * Deletes all stored files with the specified @filename from a GridFS bucket.
+   */
+  void delete_by_name(string filename);
+
+}
+```
+
+This method is an optimisation over deleting each revision of a stored file individually.
+
+**Implementation details:**
+
+Drivers MUST first find the `_id` field of all files collection documents with the given filename. Drivers MUST then
+delete all files collection documents with the found ids. Drivers MUST then delete all chunks with `files_id` in the
+found ids that was just deleted.
```

Comment:
`ids that was just deleted.` -> `ids that were just deleted.`

Or use client.bulk_write() to delete both in one command :). We can leave that as a future optimization.

Reply:
> Or use client.bulk_write() to delete both in one command :). We can leave that as a future optimization.

This would require server version 8.0 with the new client bulk write API to perform operations on both collections.